### DISCUSSION TITLE
Generating ssh key

### DISCUSSION THREAD

**Discussion by robertakid** [community/community#113537] [New to GitHub]
I have used the ssh-keygen -t ed25519 -C "my email" to try and generate a ssh key. After entering the following commands;
Enter file in which to save the key (/Users/user/.ssh/id_ed25519): /Users/user/Desktop
/Users/user/Desktop already exists.
Overwrite (y/n)? y
Enter passphrase (empty for no passphrase): 
Enter same passphrase again: 
Saving key "/Users/user/Desktop" failed: Is a directory
I am trying to save the key in a folder on the desktop of a Mac.
What am I doing wrong?

**Comment by waresnew** [comment]
> Enter **file** in which to save the key

The command asked you to provide the path to the file that you want to save the key to. You provided a path to a folder, which was Desktop, so you should also include the filename of the key (it will automatically create a file with that name). If you enter `/Users/user/Desktop/key`, then it should create a file called `key` and `key.pub` on your desktop.

**Comment by jamesleeksc** [comment]
ssh-keygen -t rsa -C "your email" -f "your name"
